source/simulator/helper_functions.cc

```diff
@@ -985,49 +985,144 @@ namespace aspect
                                           parameters.composition_degree+1);
     const QGaussLobatto<1> quadrature_formula_2 (advection_field.is_temperature() ?
                                                  parameters.temperature_degree+1 :
                                                  parameters.composition_degree+1);
-    /*
-     * TODO: currently it only works for 2 dimensional problem. Will add the 3D special selected quadrature points and weight later.
-     * Construct the quadrature points and weights:
-     * 1--Gauss points; 2-- Gauss-Lobatto points; we require that Guass-Lobatto points (2) appear  in only one direction.
-     * in 2D: the combinations are 21, 12
-     * in 3D: the combinations are 211, 121, 112
-     */
 
-    /* 2D construction
-     * Construct the quadrature points and weights:
-     * x direction uses Gauss points and y direction uses Gauss Lobatto points
-     */
-    const QAnisotropic<dim> quadrature_formula_12 (quadrature_formula_1, quadrature_formula_2);
-    //Construct the quadrature points and weights:
-    //x direction uses Gauss-Lobatto points and y direction uses Gauss points
-    const QAnisotropic<dim> quadrature_formula_21 (quadrature_formula_2, quadrature_formula_1);
-
-    const unsigned int n_q_points_12 = quadrature_formula_12.size();
-    const unsigned int n_q_points_21 = quadrature_formula_21.size();
-
-    FEValues<dim> fe_values_12 (mapping,
-                                finite_element,
-                                quadrature_formula_12,
-                                update_values   |
-                                update_quadrature_points |
-                                update_JxW_values);
-    std::vector<double> values_12 (n_q_points_12);
-
-    FEValues<dim> fe_values_21 (mapping,
-                                finite_element,
-                                quadrature_formula_21,
-                                update_values   |
-                                update_quadrature_points |
-                                update_JxW_values);
-    std::vector<double> values_21 (n_q_points_21);
+    const unsigned int n_q_points_1 = quadrature_formula_1.size();
+    const unsigned int n_q_points_2 = quadrature_formula_2.size();
+    const unsigned int n_q_points   = dim * n_q_points_2 *std::pow(n_q_points_1, dim-1) ;
+
+    std::vector< Point <dim> > points_values (n_q_points);
+
+    switch (dim)
+      {
+        case 2:
+        {
+          //make quadrature points combination 12
+          for ( unsigned int i=0; i < n_q_points_1 ; i++)
+            {
+              const Point <1>  x = quadrature_formula_1.point(i);
+              for ( unsigned int j=0; j < n_q_points_2 ; j++)
+                {
+                  const Point <1>  y = quadrature_formula_2.point(j);
+                  points_values[i * n_q_points_2+j](0) = x(0);
+                  points_values[i * n_q_points_2+j](1) = y(0);
```

Comment:
I remember that I tried it before but it caused the compiling error. I'll try again.